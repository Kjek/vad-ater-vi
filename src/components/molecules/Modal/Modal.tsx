import InputButton from '@component/atoms/Button/Button';
import IconButton from '@component/atoms/IconButton/IconButton';
import Text from '@component/atoms/Text/Text';
import { useToggle } from 'usehooks-ts';

interface ModalProps {
  variant?: 'text' | 'button';
  title: string;
  onClick?: () => void;
  children: React.ReactNode;
}

const Modal = (props: ModalProps) => {
  const { children, title, variant, onClick } = props;
  const [open, toggleOpen] = useToggle();

  const handleOnClick = () => {
    toggleOpen();
    if (onClick) {
      onClick();
    }
  };

  const toggleComponent = () => {
    switch (variant) {
      case 'text':
        return (
          <Text
            className='flex-1 cursor-pointer self-center'
            onClick={handleOnClick}
          >
            {title}
          </Text>
        );
      default:
        return (
          <InputButton
            value={title}
            className='flex-1 cursor-pointer self-center'
            onClick={handleOnClick}
          />
        );
    }
  };

  return (
    <>
      <div>
        {toggleComponent()}
        {open ? (
          <>
            <div className='absolute left-0 top-0 h-screen w-screen bg-black opacity-30' />
            <div className='dark:bg-gray-custom fixed left-[50%] top-[50%] flex max-h-[85vh] w-[90vw] max-w-[450px] translate-x-[-50%] translate-y-[-50%] flex-col gap-4 rounded-[6px] bg-white p-6 pt-10'>
              {children}
              <IconButton
                variant='cancel'
                className='absolute right-2 top-2 p-2'
                onClick={toggleOpen}
              />
            </div>
          </>
        ) : null}
      </div>
    </>
  );
};

export default Modal;
