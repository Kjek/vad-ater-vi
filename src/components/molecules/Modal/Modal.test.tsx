import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import Modal from './Modal';

describe(Modal, () => {
  it('renders correctly', () => {
    const toggleOpen = vi.fn();
    const { container } = render(
      <Modal toggleOpen={toggleOpen}>
        <div>Test</div>
      </Modal>
    );
    expect(container).toBeInTheDocument();
    expect(container).toBeVisible();
  });

  it('displays value correctly', () => {
    const toggleOpen = vi.fn();
    const { getByText } = render(
      <Modal toggleOpen={toggleOpen}>
        <button type='button' onClick={toggleOpen}>
          Toggle
        </button>
      </Modal>
    );

    const toggleButton = getByText('Toggle');
    fireEvent.click(toggleButton);
    expect(toggleOpen).toHaveBeenCalledOnce();
  });

  it('matches snapshot', () => {
    const defaultRender = render(
      <Modal toggleOpen={vi.fn()}>
        <div>Test</div>
      </Modal>
    );
    expect(defaultRender).toMatchSnapshot();
  });
});
