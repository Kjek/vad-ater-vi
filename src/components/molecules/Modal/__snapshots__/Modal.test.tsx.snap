// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fixed left-0 top-0 m-0 h-full w-full bg-black opacity-30"
      />
      <div
        class="dark:bg-gray-custom fixed left-[50%] top-[50%] flex max-h-[85vh] w-[90vw] max-w-[450px] translate-x-[-50%] translate-y-[-50%] flex-col gap-4 rounded-[6px] bg-white p-6 pt-10"
      >
        <div>
          Test
        </div>
        <button
          class="rounded-full hover:bg-gray-200 dark:hover:bg-gray-700 absolute right-2 top-2 p-2"
          type="button"
        >
          <svg
            class="text-red-500 dark:text-red-700"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.8536 2.85355C13.0488 2.65829 13.0488 2.34171 12.8536 2.14645C12.6583 1.95118 12.3417 1.95118 12.1464 2.14645L7.5 6.79289L2.85355 2.14645C2.65829 1.95118 2.34171 1.95118 2.14645 2.14645C1.95118 2.34171 1.95118 2.65829 2.14645 2.85355L6.79289 7.5L2.14645 12.1464C1.95118 12.3417 1.95118 12.6583 2.14645 12.8536C2.34171 13.0488 2.65829 13.0488 2.85355 12.8536L7.5 8.20711L12.1464 12.8536C12.3417 13.0488 12.6583 13.0488 12.8536 12.8536C13.0488 12.6583 13.0488 12.3417 12.8536 12.1464L8.20711 7.5L12.8536 2.85355Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fixed left-0 top-0 m-0 h-full w-full bg-black opacity-30"
    />
    <div
      class="dark:bg-gray-custom fixed left-[50%] top-[50%] flex max-h-[85vh] w-[90vw] max-w-[450px] translate-x-[-50%] translate-y-[-50%] flex-col gap-4 rounded-[6px] bg-white p-6 pt-10"
    >
      <div>
        Test
      </div>
      <button
        class="rounded-full hover:bg-gray-200 dark:hover:bg-gray-700 absolute right-2 top-2 p-2"
        type="button"
      >
        <svg
          class="text-red-500 dark:text-red-700"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.8536 2.85355C13.0488 2.65829 13.0488 2.34171 12.8536 2.14645C12.6583 1.95118 12.3417 1.95118 12.1464 2.14645L7.5 6.79289L2.85355 2.14645C2.65829 1.95118 2.34171 1.95118 2.14645 2.14645C1.95118 2.34171 1.95118 2.65829 2.14645 2.85355L6.79289 7.5L2.14645 12.1464C1.95118 12.3417 1.95118 12.6583 2.14645 12.8536C2.34171 13.0488 2.65829 13.0488 2.85355 12.8536L7.5 8.20711L12.1464 12.8536C12.3417 13.0488 12.6583 13.0488 12.8536 12.8536C13.0488 12.6583 13.0488 12.3417 12.8536 12.1464L8.20711 7.5L12.8536 2.85355Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
